### DISCUSSION TITLE
asdf issues - would this solution break Homebrew’s policies?

### DISCUSSION THREAD

**Discussion by andrewcrook** [Homebrew/discussions#3720] [Writing Formulae/Casks]
Issue https://github.com/asdf-vm/asdf/issues/1103#issuecomment-1093094933

I suggested on https://github.com/asdf-vm/asdf/issues/1103#issuecomment-1263735629

The issue is any shims made by asdf have paths which break upon a homebrew upgrade because of the use of version numbers in paths under the Cellar.

> I don't know why the Homebrew formula doesn’t just create symbolic links to `~/.asdf` from `/opt/homebrew/Cellar/asdf/<version>/` 
> that would maintain asdf’s file structure and paths in configuration and shims allowing homebrew to keep the application under` homebrew/Celler`. 

**But would this break one of homebrew's own standards?**

**Comment by jfly** [comment]
> just create symbolic links to ~/.asdf from /opt/homebrew/Cellar/asdf/<version>/

Do you mean this the other way around, @andrewcrook? I don't know why we'd want to have symlinks that point to `~/.asdf`, which may not even exist.

Assuming you mean symlinks from `~/.asdf` to `/opt/homebrew/Cellar/asdf/<version>/`, I *personally* would be surprised if brew touched anything in my home directory. I'd only expect it to touch files under `/opt/homebrew`.

I am hopeful that the (long overdue) https://github.com/Homebrew/homebrew-core/pull/112102 will address all the confusion this has caused people.

**Reply by andrewcrook** [reply]
> Assuming you mean symlinks from ~/.asdf to /opt/homebrew/Cellar/asdf/<version>/, 

@jfly Yes, sorry I always get that the wrong way around

by default asdf installs into ~/.asdf 

I have seen similar issues in the past with formulas, such as poetry, that end up changing the original scripts and therefore adds an extra element that has to be maintained. 

I just thought the symbolic links would be simpler. I am surprised it hasn’t been used or suggested so I wondered what the reason was  against it?  that’s all.

> I am hopeful that the (long overdue) https://github.com/Homebrew/homebrew-core/pull/112102 will address all the confusion this has caused people.

Look forward to it.
